integrated_channels/canvas/client.py

```diff
@@ -51,9 +51,102 @@ def __init__(self, enterprise_configuration):
+                course_id = course['id']
+                break
+
+        if not course_id:
+            raise CanvasClientError(
+                "Course: {course_id} not found registered in Canvas for Edx learner: {user_id}"
+                "/Canvas learner: {canvas_user_id}.".format(
+                    course_id=learner_data['courseID'],
+                    user_id=learner_data['userID'],
+                    canvas_user_id=canvas_user_id
+                ))
+
+        # Check if the course assignment already exists
+        canvas_assignments_url = '{canvas_base_url}/api/v1/courses/{course_id}/assignments'.format(
+            canvas_base_url=self.enterprise_configuration.canvas_base_url,
+            course_id=course_id
+        )
+        assignments_resp = self.session.get(canvas_assignments_url).json()
+        assignment_id = None
+        for assignment in assignments_resp:
+            if assignment['integration_id'] == integration_id:
+                assignment_id = assignment['id']
+                break
+
+        # Canvas requires a course assignment for a learner to be assigned a grade.
+        # If no assignment has been made yet, create it.
+        if not assignment_id:
+            assignment_creation_data = {
+                'assignment': {
+                    'name': 'Final Grade',
+                    'submission_types': 'none',
+                    'integration_id': integration_id,
+                    'published': True,
+                    'points_possible': 100
+                }
+            }
+            create_assignment_resp = self.session.post(canvas_assignments_url, json=assignment_creation_data)
+            assignment_id = create_assignment_resp.json()['id']
+
+        # Post a grade for the assignment. This shouldn't create a submission for the user, but update the grade.
+        submission_url = '{base_url}/api/v1/courses/{course_id}/assignments/' \
+                         '{assignment_id}/submissions/{user_id}'.format(
+                             base_url=self.enterprise_configuration.canvas_base_url,
+                             course_id=course_id,
+                             assignment_id=assignment_id,
+                             user_id=canvas_user_id
+                         )
+
+        # The percent grade from the grades api is represented as a decimal
+        submission_data = {
+            'submission': {
+                'posted_grade': learner_data['grade'] * 100
+            }
+        }
+        update_grade_response = self.session.put(submission_url, json=submission_data)
+        update_grade_response.raise_for_status()
+        return update_grade_response.status_code, update_grade_response.text
 
     def delete_course_completion(self, user_id, payload):  # pylint: disable=unused-argument
+        # Wondering how we want to best delete the user's course completion data. We could potentially un-enroll
+        # the student from the course here. Is that sufficient?
```

Comment:
Just want to double-check:
- Are these comments for the purposes of reviewers or do you intend to leave them in the code? 
  - If for the reviewers, I'd recommend just "reviewing" (ie just commenting) in the PR on relevant lines yourself before assigning the PR to someone. You can tag them in questions or let them answer as they come across it.
  - If you did intend to leave these in the code for future devs or what have you: don't?

With squashing it's not a huge deal, but it seems silly to have to basically revert/uncommit these comments when it could just be in the PR.

Reply:
The comments are for the reviewers, yes. I've had some questions pop up along the process and thought it would be useful to leave them for discussion. I'm going to delete the comment after they've been resolved/discussed.

Reply:
I think just commenting on the line in the PR would be ideal. I guess in this case there was no change to this method, so it wouldn't have shown up, but in general I'm not a huge fan of modifying code to basically use it as a discussion board.

The plan is to remove it, but it could get forgotten and as these PRs are public, I'd prefer not having code changes someone might be looking at that are just unnecessary comments. I think given there was no work done on this method and you had concerns about its inclusion in this scope, this probably warranted a slack discussion/call/etc. instead.

Reply:
fair points, I can transfer to comments in the PR.